### DISCUSSION TITLE
Install w/o root permission and w/o internet access

### DISCUSSION THREAD

**Discussion by smprather** [LunarVim/LunarVim#4158] [Q&A] [answered]
Hi Lunar People--

I am trying to install on our internal system for which I cannot a) sudo, b) see the external internet. I can, however, transfer arbitrary files in (tar.gz, etc). The system is RH8. uname -s responds as Linux. I have nvim/0.9.0, npm/8.19.3, pip/python/3.11 and cargo/1.68.1 available. I transferred the lvim tar.gz into the closed system. But when I look at the install script, I see lots of sudo. Is what I'm trying to do possible? If not, can it be made possible? Thanks!

```bash
$ ./utils/installer/install.sh --local --no-install-dependencies

      88\                                                   88\
      88 |                                                  \__|
      88 |88\   88\ 888888$\   888888\   888888\ 88\    88\ 88\ 888888\8888\
      88 |88 |  88 |88  __88\  \____88\ 88  __88\\88\  88  |88 |88  _88  _88\
      88 |88 |  88 |88 |  88 | 888888$ |88 |  \__|\88\88  / 88 |88 / 88 / 88 |
      88 |88 |  88 |88 |  88 |88  __88 |88 |       \88$  /  88 |88 | 88 | 88 |
      88 |\888888  |88 |  88 |\888888$ |88 |        \$  /   88 |88 | 88 | 88 |
      \__| \______/ \__|  \__| \_______|\__|         \_/    \__|\__| \__| \__|

--------------------------------------------------------------------------------
Detecting platform for managing any additional neovim dependencies
Error detected while processing command line:
E15: Invalid expression: <hangs here>
^C
```

**Comment by LostInTheLogs** [comment, marked as answer]
.tar.gz from here should work https://github.com/LunarVim/LunarVim-mono/releases/tag/master

**Reply by smprather** [reply]
The appImage works for me. Thanks @LostNeophyte !

**Reply by smprather** [reply]
Well, I spoke too soon. When opening a python file, it tries to install pyright.
![image](https://github.com/LunarVim/LunarVim/assets/6503931/5623222e-3964-4fff-9d97-9cc441d8a61e)

**Reply by smprather** [reply]
Same result with tar.gz.

**Reply by LostInTheLogs** [reply]
that should be fixed, open an issue, will fix it later

**Reply by LostInTheLogs** [reply]
https://github.com/LunarVim/LunarVim/issues/4162

**Reply by smprather** [reply]
I created a venv and installed everything pynvim, neovim, pyright. I tried lvim from that activated venv. Same result. Thanks again!

In the meantime, I installed lvim in my WSL instance that can see the internet. Then I copied all of the relevant ~/.local/share over to my secure system. That is working ok for me. CheckHealth looks reasonably ok (despite all the hangs trying to run curl to check for latest versions of stuff).
